kyo-core/shared/src/test/scala/kyoTest/ResultTest.scala

```diff
@@ -0,0 +1,285 @@
+        }
+    }
+
+    "pattern matching" - {
+        "deep matching" - {
+            val nestedResult = Success(Success(Success(Success(1))))
+
+            "should match deeply nested Success and extract inner value" in {
+                val result = nestedResult match
+                    case Success(Success(Success(Success(x)))) => x
+                assert(result == 1)
+            }
+
+            "should match partially and extract nested Result" in {
+                val result = nestedResult match
+                    case Success(Success(x)) => x
+                assert(result == Success(Success(1)))
+            }
+        }
+
+        "matching with guards" - {
+            "should match Success with a guard" in {
+                val tryy: Result[Int] = Success(2)
+                val result = tryy match
+                    case Success(x) if x > 1 => "greater than 1"
+                    case Success(_)          => "less than or equal to 1"
+                    case Failure(_)          => "failure"
+                assert(result == "greater than 1")
+            }
+
+            "should match Failure with a guard" in {
+                val tryy: Result[Int] = Failure(new Exception("error"))
+                val result = tryy match
+                    case Failure(e) if e.getMessage.length > 5 => "long error"
+                    case Failure(_)                            => "short error"
+                    case Success(_)                            => "success"
+                assert(result == "short error")
+            }
+        }
+
+        "matching nested Results" - {
+            val try1: Result[Result[Int]] = Success(Success(1))
+            val try2: Result[Result[Int]] = Success(Failure(new Exception("error")))
+            val try3: Result[Result[Int]] = Failure(new Exception("error"))
+
+            "should match deeply nested Success" in {
+                val result = try1 match
+                    case Success(Success(x)) => x
+                    case _                   => 0
+                assert(result == 1)
+            }
+
+            "should match Success containing Failure" in {
+                val result = try2 match
+                    case Success(Failure(e)) => e.getMessage
+                    case _                   => ""
+                assert(result == "error")
+            }
+
+            "should match top-level Failure" in {
+                val result = try3 match
+                    case Failure(e) => e.getMessage
+                    case _          => ""
+                assert(result == "error")
+            }
+        }
+    }
+
+    "edge cases" - {
+        "should handle exceptions during fold" in {
+            val tryy      = Success(1)
+            val exception = new RuntimeException("exception")
+            val result =
+                try
+                    tryy.fold(_ => throw exception, _ => throw exception)
```

Comment:
https://github.com/scala/scala/blob/2.13.x/src/library/scala/util/Try.scala#L288-L289

I don't think this is the behavior most users of `Try` would expect.

Reply:
good catch! fixed